pkg/controller/factory/factory.go

```diff
@@ -16,14 +16,21 @@ type Factory struct {
 	sync                  SyncFunc
 	resyncInterval        time.Duration
 	objectQueue           bool
-	informers             []cache.SharedInformer
+	informers             []RegistrableInformer
 	namespaceInformers    []*namespaceInformer
 	cachesToSync          []cache.InformerSynced
 	interestingNamespaces sets.String
 }
 
+// RegistrableInformer represents any structure that allow to register event handlers and informs if caches are synced.
+// Any SharedInformer will comply.
+type RegistrableInformer interface {
```

Comment:
this isn't part of the factory library?

Reply:
it is... it is a subset of SharedInformer interface so the "Deployer" thing can comply... We don't need full shared informer access... just event handler registration and has synced.